src/components/GitStash.tsx

```diff
@@ -0,0 +1,386 @@
+  //     await props.model.stash_apply(index);
+  //   } catch (err) {
+  //     console.error(err);
+  //   }
+  // };
+
+  const getFilePath = (file: string) => {
+    // Root directory
+    if (!file.includes('/')) {
+      return {
+        name: file,
+        path: ''
+      };
+    } else {
+      // One folder
+      const fileSplit = file.split('/');
+      if (fileSplit.length == 2) {
+        return {
+          path: fileSplit[0],
+          name: fileSplit[1]
+        };
+        // More than one nested folder
+      } else {
+        return {
+          name: fileSplit.pop(),
+          path: fileSplit.join('/')
+        };
+      }
+    }
+  };
+
+  return (
+    <div className={sectionFileContainerStyle}>
+      <div
+        className={sectionAreaStyle}
+        onClick={() => {
+          if (props.collapsible && props?.files.length > 0) {
+            setShowStashFiles(!showStashFiles);
+          }
+        }}
+      >
+        {props.collapsible && (
+          <button className={changeStageButtonStyle}>
+            {showStashFiles && props?.files.length > 0 ? (
+              <caretDownIcon.react />
+            ) : (
+              <caretRightIcon.react />
+            )}
+          </button>
+        )}
+        <span
+          className={sectionHeaderLabelStyle}
+          style={{ display: 'flex', justifyContent: 'space-between' }}
+        >
+          <p>
+            {props.message} (on {props.branch})
+          </p>
+          <span style={{ display: 'flex' }}>{props.actions}</span>
+        </span>
+      </div>
+      {showStashFiles && (
+        <FixedSizeList
+          height={Math.max(
+            Math.min(props.height - HEADER_HEIGHT, nFiles * ITEM_HEIGHT),
+            ITEM_HEIGHT
+          )}
+          itemCount={nFiles}
+          itemData={props?.files}
+          itemKey={(index, data) => data[index]}
+          itemSize={ITEM_HEIGHT}
+          style={{ overflowX: 'hidden' }}
+          width={'auto'}
+        >
+          {({ index, style }) => {
+            const file = props.files[index];
+            const { name, path } = getFilePath(file);
+
+            return (
+              <div style={{ display: 'flex', padding: '0 4px' }}>
+                <fileIcon.react
+                  className={fileIconStyle}
+                  elementPosition="center"
+                  tag="span"
+                />
+                <span className={fileLabelStyle}>
+                  {name}
+                  <span
+                    // style={{
+                    //   fontSize: 'var(--jp-ui-font-size0)',
+                    //   margin: '0 4px',
+                    //   color: 'var(--jp-ui-font-color2)'
+                    // }}
+                    className={folderLabelStyle}
+                  >
+                    {' '}
+                    <span>{path}</span>
+                  </span>
+                  {/* <span className={folderLabelStyle}>{folder}</span> */}
+                </span>
```

Comment:
Could you reuse `FilePath`?

Reply:
Sure. I don't have the filetype props though, so I just left filetype as null.